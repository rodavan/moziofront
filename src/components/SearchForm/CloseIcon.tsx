import React from "react";

const CloseIcon: React.FC = () => {
  return (
    <svg
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      data-testid="close-icon"
    >
      <path
        d="M6 18L18 6M6 6L18 18"
        stroke="#7786D2"
        strokeWidth="2"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
};

export default CloseIcon;
