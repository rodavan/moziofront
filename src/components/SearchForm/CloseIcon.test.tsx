import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import CloseIcon from "./CloseIcon";

describe("CloseIcon", () => {
  test("renders CloseIcon component", () => {
    const { asFragment } = render(<CloseIcon />);
    const closeIcon = screen.getByTestId("close-icon");
    expect(closeIcon).toBeInTheDocument();

    // Snapshot testing
    expect(asFragment()).toMatchSnapshot();
  });
});
