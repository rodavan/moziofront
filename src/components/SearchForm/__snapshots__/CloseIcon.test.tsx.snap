// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseIcon renders CloseIcon component 1`] = `
<DocumentFragment>
  <svg
    data-testid="close-icon"
    fill="none"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 18L18 6M6 6L18 18"
      stroke="#7786D2"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</DocumentFragment>
`;
